docs/src/configuration/app/hooks/_index.md

```diff
@@ -0,0 +1,258 @@
+Do these tasks using one of [three hooks](./hooks-comparison.md).
+
+The following example goes through each of these scenarios for the [Gatsby Drupal template](https://github.com/platformsh-templates/gatsby-drupal).
+This template uses [Drupal](https://www.drupal.org/) as the headless CMS backend
+and [Gatsby](https://www.gatsbyjs.com/) for the frontend.
+To get the code ready, you need to run a few commands.
+The example commands are somewhat simplified, but you can find them all in the [GitHub repository](https://github.com/platformsh-templates/gatsby-drupal).
+
+In this case, you have two applications and so two [`.platform.app.yaml` configuration files](../_index.md).
+Each file is in the folder for that application.
+You run two hooks in the file for Drupal and one hook in the file for Gatsby.
+
+## Compile extension
+
+For this template to work, it needs a [PHP extension for Redis](https://github.com/phpredis/phpredis).
+This includes getting the up-to-date code for the extension and then compiling it.
+
+This action needs to write to disk and doesn't need any relationships with other services.
+This makes it perfect for a `build` hook.
+
+The exact details of the extension aren't so important to the hook,
+so create a bash script file to handle it for you.
+
+1. Copy the [script from the template](https://github.com/platformsh-templates/gatsby-drupal/blob/master/drupal/install-redis.sh)
+   into a file called `install-redis.sh`.
+   Note that hooks are executed using the dash shell, not the bash shell used by SSH logins.
+1. Create a `build` hook in your [app configuration](../app-reference.md):
+
+   ```yaml {location="drupal/.platform.app.yaml"}
+   hooks:
+       build: |
+           set -e
+   ```
+
+   There are two parts to the hook so far:
+
+     * The `|` means the lines that follow can contain a series of commands.
+       They aren't interpreted as new YAML properties.
+     * Adding `set -e` means that the hook fails if _any_ of the commands in it fails.
+       Without this setting, the hook fails only if its _final_ command fails.
+
+       If a `build` hook fails for any reason, the build is aborted and the deploy doesn't happen.
+       Note that this only works for `build` hooks.
+       If other hooks fail, the deploy still happens.
+1. Call your script from the `build` hook:
+
+   ```yaml {location="drupal/.platform.app.yaml"}
+   hooks:
+       build: |
+           set -e
+           bash install-redis.sh 4.3.0
+   ```
+
+   This call includes the version of the Redis extension to compile.
+
+Now the extension is compiled every time the Drupal app is built.
+
+## Configure Drush
+
+The template uses [Drush](https://www.drush.org/latest/) to handle routine tasks.
+For its configuration, Drush needs the URL of the site.
+That means the configuration can't be done in the `build` hook, when the site isn't yet deployed and so the URL is unknown.
```

Comment:
```suggestion
That means the configuration can't be done in the `build` hook, when the site isn't yet deployed, because the URL is unknown at this stage.
```

Reply:
This suggestion shows that the original sentence is unclear. I wouldn't take the suggestion because the cause and effect are supposed to be the other way around (no deploy → no URL, not no URL → no deploy), but I will definitely try to clarify it.